Orange/widgets/data/owcorrelations.py

```diff
@@ -232,7 +232,7 @@ def _set_empty_status(self):
         self.master.setStatusMessage("")
 
 
-class OWCorrelations(OWWidget):
+class OWCorrelations(OWWidget, openclass=True):
```

Comment:
You mustn't do that. This flag is not here to be arbitrarily overridden.

By setting this to `True`, you promise that this widget will not make any backward-incompatible changes to its non-private methods and attributes. We never do this in "final" widgets.

(I just discovered `OWPreprocess` does it, but I believe this is the only exception. There, it's done to support preprocessing for text, I believe; the two are closely related and will be changed in tandem. But when we change Correlations, nobody will think about Interactions or want to maintain the two together.)

We have had a conglomerate of Mosaic, Sieve and perhaps a few others importing functions (that should be methods) from each other. After cleaning up that mess, no widget module does (well, at least no widget module should) import another widget module.

As it looks currently, Interactions is hacked atop of Correlations, though Correlations is not designed to be used that way.

The proper was to do this is to extract methods that are common to Correlations and your widget into another, base widget and put it into a separate file. Only then can you set the `openclass` flag (of the base widget) to `True`.

Also, think about what this base class would mean semantically. Finding a good name for it, for instance, would be a step toward it.

Reply:
Thank you for pointing this out, I will remove it and only keep it in my local code as I continue working on Interactions.

Reply:
@janezd Just a comment. Preprocess Text does not depend on Preprocess (the pipeline is vastly different). Perhaps it has something to do with Spectoscopy, but just a note - it should be investigated.

Reply:
Thanks for this comment, it demonstrates the problem really well. All I know is that `OWPreprocess` is open and that @VesnaT opened it. Perhaps I knew why and the reasoning was sound and I agreed with it ... but as things stand now, we don't know who relies on it and what will fail if we decide to change something.

... two minutes later: I discovered I merged this, here https://github.com/biolab/orange3/pull/4521. We had good reasons, but I regret it, we should have refactored it. SCOrange can explode at any change of Preprocess without anybody noticing.

Reply:
Emm, sorry, single cell.